1.regression_modelling/scripts/0.split_data.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from pycytominer import feature_select
+
+
+# ## Load in normalized profiles and concat into one dataframe
+
+# In[2]:
+
+
+# path to big drive where data is located (large)
+drive_path = pathlib.Path("../../../../media/18tbdrive/")
+
+# set path to folder in image profiling repo with the normalized profiles
+norm_path = pathlib.Path(
+    drive_path / "Github_Repositories/nuclear_speckle_image_profiling/4.preprocess_features/data/single_cell_profiles"
+).resolve(strict=True)
+
+# load all normalized parquet files
+files = norm_path.glob("*_sc_normalized.parquet")
+dfs = [pd.read_parquet(file) for file in files]
+
+# concatenate them into one data frame
+combined_df = pd.concat(dfs, ignore_index=True)
+
+# perform feature selection to drop any columns that have NaN (avoid downstream issues)
+combined_df = feature_select(
+    combined_df,
+    operation="drop_na_columns",
+    na_cutoff=0
+)
+
+# print df
+print(combined_df.shape)
+combined_df.head()
+
+
+# ## Load in filtered single-cell data from first module
+
+# In[3]:
+
+
+# load in filtered tuple dataframe
+filtering_df = pd.read_parquet(
+    pathlib.Path(
+        "../0.data_analysis_and_processing/filtered_single_cells/filtered_single_cell_profiles.parquet"
+    )
+)
```

Comment:
What are the filtered sc? Are these QC'd? How is this df different from the feature selected one?

A markdown cell at the top of the notebook explaining the "sub-sections" of this notebook might help

Reply:
I can add a quick snippet of information at the top of the file!

To explain here, for the methodology that Cameron will use with the images, he needs to clear out more single-cells that are out of a specific range of width and height for the crop. This is what is known as the "filtering_df". 

To make sure that both of the methods (mine and Cameron's) is comparable, we have to use the same single-cells which is why we have a filtering dataframe to filter down the morphological profiles.

Reply:
Ahhhh, I see. This makes sense. Documentation would help.

I think a small figure in a future PR showing an example of a non-square "filtered out" crop would be great